src/OrchardCore.Modules/OrchardCore.Contents/Indexing/FullTextContentIndexHandler.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.IO;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Fluid;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata;
+using OrchardCore.ContentManagement.Metadata.Settings;
+using OrchardCore.ContentManagement.Models;
+using OrchardCore.Indexing;
+using OrchardCore.Liquid;
+
+namespace OrchardCore.Contents.Indexing
+{
+    public class FullTextContentIndexHandler : IContentItemIndexHandler
+    {
+        private readonly IContentDefinitionManager _contentDefinitionManager;
+        private readonly ILiquidTemplateManager _liquidTemplateManager;
+        public const string FullTextKey = "Content.ContentItem.FullText";
+
+        public FullTextContentIndexHandler(
+            IContentDefinitionManager contentDefinitionManager,
+            ILiquidTemplateManager liquidTemplateManager)
+        {
+            _contentDefinitionManager = contentDefinitionManager;
+            _liquidTemplateManager = liquidTemplateManager;
+        }
+
+        public async Task BuildIndexAsync(BuildIndexContext context)
+        {
+            var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(context.ContentItem.ContentType);
+
+            if (contentTypeDefinition == null)
+            {
+                return;
+            }
+
+            var settings = contentTypeDefinition.Settings.ToObject<ContentTypeSettings>();
+
+            if (settings.IsFullText)
+            {
+                var result = String.Empty;
+                if (!String.IsNullOrEmpty(settings.FullText))
+                {
+                    var templateContext = new TemplateContext();
+                    templateContext.SetValue("Model", context.ContentItem);
+
+                    using (var writer = new StringWriter())
```

Comment:
Pretty sure there is a custom extension method that returns the string directly without you having to create a string writer so it can reuse a pooled string builder.

Reply:
I need to find it. Let me know if you remember where it is. This is the first thing I tried to do but cut the chase to using it directly to see if it was working 😄

Reply:
Done